Enhanced and customizable reject reason (#1170)
Haivision/srt: docs/API-functions.md

The `SRTO_PACKETFILTER` option has been set differently on both connection
parties.

> Beside these codes, there could be also server and user rejection codes,
> as defined by `SRT_REJC_SYSTEM`, `SRT_REJC_SERVER` and `SRT_REJC_USER`
> constants. These above values are all below `SRT_REJC_SERVER` value.
> From this value on, the number space is reserved for "server codes"
> (`SRT_REJC_SERVER` value plus HTTP codes) and values above `SRT_REJC_USER`
> are freely defined by the application. In addition to this code, the
> extended free-form rejection message can be obtained by reading the
> value of `SRTO_STREAMID` value on a socket that received the rejection.

Reviewer: ```suggestion
There may also be server and user rejection codes,
as defined by the `SRT_REJC_SYSTEM`, `SRT_REJC_SERVER` and `SRT_REJC_USER`
constants. Note that the number space from the value of `SRT_REJC_SERVER`  and above is reserved for "server codes" (`SRT_REJC_SERVER` value plus HTTP codes). Values above `SRT_REJC_USER`
are freely defined by the application. In addition to this code, the
extended free-form rejection message can be obtained by reading the
value of `SRTO_STREAMID` on a socket that received the rejection.
```

```

Returns a constant string for the reason of the connection rejected,

> as per given code id. It provides a system-defined message for
> values below `SRT_REJC_SERVER` and for values from the server and
> user ranges just the number with "SERVER" or "USER" prefix.

Reviewer: ```suggestion
as per given code ID. It provides a system-defined message for
values below `SRT_REJC_SERVER`. For values from the server and
user ranges it provides just the number with the prefix "SERVER" or "USER".
```

The actual messages assigned to system rejection codes, that is,
between `SRT_REJC_SYSTEM` and `SRT_REJ_E_SIZE` (both exclusive), can be also
obtained from `srt_rejectreason_msg` array.

### srt_setrejectreason

```
int srt_setrejectreason(SRTSOCKET sock, int value);
```

Sets the rejection code on the socket. This call is only useful in the
listener callback. The code from `value` set this way will be set as a

> rejection reason for the socket and after the callback decided to reject
> the connection the code will be passed back to the caller peer with the

Reviewer: ```suggestion
rejection reason for the socket. After the callback rejects
the connection, the code will be passed back to the caller peer with the
```

handshake response.

> Note that allowed values for that function begin with `SRT_REJC_SERVER`
> (that is, you cannot set a system rejection code using that function).
> For example, your application can inform the calling side that the resource
> specified under the `r` key in the StreamID string is not availble - it
> sets then the value `SRT_REJC_SERVER + 404`.

Reviewer: ```suggestion
Note that allowed values for this function begin with `SRT_REJC_SERVER`
(that is, you cannot set a system rejection code using` srt_setrejectreason(SRTSOCKET sock, int value);`).
For example, your application can inform the calling side that the resource
specified under the `r` key in the StreamID string is not availble - it
then sets the value to `SRT_REJC_SERVER + 404`.
```
